1. Импортируем датасет и проведем необходимые операции со столбцами
2. Выберем 20 человек с ростом от 170 до 180 см.
3. Посчитаем среднее значение веса людей из всего набора данных и из нашей выборки и определим, является ли разница статистически значимой

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import norm

%matplotlib inline

In [2]:
data = pd.read_csv( 'hw_25000.csv', names = [ 'height_inches', 'weight_pounds'], header = 0 )
data['height'] = data['height_inches'] * 2.54
del data['height_inches']
data.head()

weight_pounds      height
1       112.9925  167.089607
2       136.4873  181.648633
3       153.0269  176.272800
4       142.3354  173.270164
5       144.2971  172.181037

In [3]:
mydata = data.loc[(data['height'] >= 170) & (data['height'] <= 180)  ]
mydata = mydata.head(20)
mydata = mydata.reset_index(drop = True)

In [4]:
data['weight_pounds'].mean(), mydata['weight_pounds'].mean()

(127.07942116080001, 132.92411)

Среднее значение веса из "выборки" больше, чем среднее значение веса из всего объема данных.
Проверим, является ли эта разница статистически значимой.
Вычислим P-значение по T-критерию Стьюдента с уровнем значимости 0.05 (исходя из предположения, что мы получили число случайную выборку из 20 строк лишь с 5% вероятностью)

In [5]:
sample = mydata['weight_pounds'].tolist()

In [6]:
stats.ttest_1samp( sample, data['weight_pounds'].mean())

Ttest_1sampResult(statistic=2.2438433326710006, pvalue=0.03695275215128037)

P-значение равно 0.037, оно меньше, чем уровень значимости, следовательно у нас есть основание отклонить нуль-гипотезу и мы можем утверждать, что разница среднего веса людей в исходном файле и среднего веса людей в выборке является статистически значимой.